contribs/gnogenesis/internal/txs/txs_add_packages_test.go

```diff
@@ -60,66 +63,236 @@ func TestGenesis_Txs_Add_Packages(t *testing.T) {
 		assert.ErrorContains(t, cmdErr, errInvalidPackageDir.Error())
 	})
 
-	t.Run("invalid deployer address", func(t *testing.T) {
+	t.Run("nonExistentKeyName", func(t *testing.T) {
 		t.Parallel()
+		keybaseDir := t.TempDir()
+		keyname := "beep-boop"
 
 		tempGenesis, cleanup := testutils.NewTestFile(t)
 		t.Cleanup(cleanup)
 
 		genesis := common.GetDefaultGenesis()
 		require.NoError(t, genesis.SaveAs(tempGenesis.Name()))
 
+		io := commands.NewTestIO()
+		io.SetIn(
+			strings.NewReader(
+				fmt.Sprintf(
+					"%s\n",
+					"password",
+				),
+			),
+		)
 		// Create the command
-		cmd := NewTxsCmd(commands.NewTestIO())
+		cmd := NewTxsCmd(io)
 		args := []string{
 			"add",
 			"packages",
 			"--genesis-path",
 			tempGenesis.Name(),
 			t.TempDir(), // package dir
-			"--deployer-address",
-			"beep-boop", // invalid address
+			"--key-name",
+			keyname, // non-existent key name
+			"--gno-home",
+			keybaseDir, // temporaryDir for keybase
+			"--insecure-password-stdin",
 		}
 
 		// Run the command
 		cmdErr := cmd.ParseAndRun(context.Background(), args)
-		assert.ErrorIs(t, cmdErr, errInvalidDeployerAddr)
+		fmt.Println(cmdErr.Error())
+		assert.ErrorContains(t, cmdErr, "Key "+keyname+" not found")
 	})
 
-	t.Run("valid package", func(t *testing.T) {
+	t.Run("existentKeyBadPassword", func(t *testing.T) {
 		t.Parallel()
 
 		tempGenesis, cleanup := testutils.NewTestFile(t)
 		t.Cleanup(cleanup)
 
 		genesis := common.GetDefaultGenesis()
 		require.NoError(t, genesis.SaveAs(tempGenesis.Name()))
+		// Prepare the package
+		var (
+			packagePath = "gno.land/p/demo/cuttlas"
+			dir         = t.TempDir()
+			keybaseDir  = t.TempDir()
+			keyname     = "beep-boop"
+			password    = "somepass"
+		)
+		createValidFile(t, dir, packagePath)
+		// Create key
+		kb, err := keys.NewKeyBaseFromDir(keybaseDir)
+		require.NoError(t, err)
+		mnemonic, err := client.GenerateMnemonic(256)
+		require.NoError(t, err)
+		_, err = kb.CreateAccount(keyname, mnemonic, "", password+"wrong", 0, 0)
+		require.NoError(t, err)
+
+		io := commands.NewTestIO()
+		io.SetIn(
+			strings.NewReader(
+				fmt.Sprintf(
+					"%s\n",
+					password,
+				),
+			),
+		)
+
+		// Create the command
+		cmd := NewTxsCmd(io)
+		args := []string{
+			"add",
+			"packages",
+			"--genesis-path",
+			tempGenesis.Name(),
+			"--key-name",
+			keyname, // non-existent key name
+			"--gno-home",
+			keybaseDir, // temporaryDir for keybase
+			"--insecure-password-stdin",
+			dir,
+		}
+
+		// Run the command
+		cmdErr := cmd.ParseAndRun(context.Background(), args)
+		fmt.Println(cmdErr.Error())
+		assert.ErrorContains(t, cmdErr, "unable to sign txs, unable sign tx invalid account password")
```

Comment:
Can we improve this check?
This is super flaky, checking for a direct string

Reply:
is this better ?
cfcc35d1e88d4afed936fa4b69a13e325908ea8f